app/policies/user_policy.rb

```diff
@@ -11,6 +11,6 @@ def show_current_user?
   end
 
   def access_private_data?
-    current_user == resource
+    !!current_user && current_user.id == resource.id
```

Comment:
Redundant `!!`:
`&&` will always return `true` or `false`.

Reply:
Hm I had a problem that it returned `nil` instead of `false` when `current_user` was `nil`; will check again.

Reply:
```ruby
nil && 4 == 5
=> nil
```

Reply:
Ah okay. Sorry for misleading you.